game/player.cc

```diff
@@ -0,0 +1,47 @@
+#include <vector>
+#include <memory>
+#include <utility>
+
+#include "player.h"
+#include "hackmon.h"
+#include "statInfo.h"
+using namespace std;
+
+Player::Player(string name, vector<unique_ptr<Hackmon>> party): name{name}, party{party}, winTotal{0} {
+  nextAlive = numberBattling; // FIXME: global variable
+
+  for (auto& index : numberBattling) {
```

Comment:
You can't iterate over `numberBattling` because it's an int. Probably easiest to just use a regular for loop

Reply:
```suggestion
  for (size_t index = 0; index < numberBattling; ++index) {
```